apps/hash-graph/lib/graph/src/api/rest.rs

```diff
@@ -111,6 +111,23 @@ pub struct RestRouterDependencies<P: StorePool + Send + 'static> {
     pub domain_regex: DomainValidator,
 }
 
+/// A [`Router`] that only serves the OpenAPI specification (JSON, and necessary subschemas) for
+/// the REST API.
+pub fn openapi_only_router() -> Router {
+    let open_api_doc = OpenApiDocumentation::openapi();
+
+    Router::new()
+        .route(
+            "/api-doc/openapi.json",
+            get({
+                let doc = open_api_doc;
+                move || async { Json(doc) }
+            }),
+        )
+        .route("/api-doc/models/*path", get(serve_static_schema))
```

Comment:
```suggestion
    Router::new().nest(
        "/api-doc",
        Router::new()
            .route(
                "/openapi.json",
                get({
                    let doc = open_api_doc;
                    move || async { Json(doc) }
                }),
            )
            .route("/models/*path", get(serve_static_schema)),
    )
```

Reply:
[Unify nesting of OpenAPI spec and fix bug](https://github.com/hashintel/hash/pull/2358/commits/7b713ae80319496cc1626f576a5006a48d5179d3)